libraries/USBHost/USBHost/USBEndpoint.cpp

```diff
@@ -140,7 +140,7 @@ void USBEndpoint::queueTransfer()
 
     //Now add this free TD at this end of the queue
     state = USB_TYPE_PROCESSING;
-    td_current->nextTD = td_next;
+    td_current->nextTD = (hcTd*)td_next;
```

Comment:
Thank you your comments.

IAR compiler was an error of the following for the following line.
  libraries\USBHost\USBHost\USBHostTypes.h (146)
  ---------------------------
  [Error][Pe256]: invalid redeclaration of type name "HCTD" (declared at line 139)
    } PACKED HCTD;
                       ^
  ---------------------------

we have changed to not use the same symbol. 
Then, IAR compiler was an error of the following for the following line.
  libraries\USBHost\USBHost\USBEndpoint.cpp (143)
  libraries\USBHost\USBHost\USBEndpoint.cpp (161)
  libraries\USBHost\USBHost\USBHost.cpp (307)
  ---------------------------
  [Error][Pe513]: a value of type "hcTd volatile __packed *" cannot be assigned to an entity of type "hcTd volatile *"
        td_current->nextTD = td_next;
                                        ^
  ---------------------------

Therefore, we have changed the code against these errors.
We did a pull request these contents.
In addition, It is a confirmed operation even ARMCC and GCC ARM.